geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/AbstractFileSystemStorage.scala

```diff
@@ -71,7 +71,7 @@ abstract class AbstractFileSystemStorage(
     * @param observer observer to report stats on the data written
     * @return
     */
-  protected def createWriter(file: Path, observer: FileSystemObserver): FileSystemWriter
+  protected def createWriter(file: Path, observer: BoundsObserver): FileSystemWriter
```

Comment:
i think this needs to stay as FileSystemObserver here, b/c the user can specify custom observers at runtime. we should probably make FileSystemWriter essentially be it's own observer for bounds, and then we can change the bounds observer here to just pull the bounds out of the writer with a callback.

Reply:
the way you have it now does kind of work, but seems a bit confusing and a mixing of concerns. since the writer now *requires* tracking its own bounds, it should just do that itself.